include/rdma/fi_eq.h

```diff
@@ -239,6 +239,8 @@ struct fi_cq_err_entry {
 	/* err_data is available until the next time the CQ is read */
 	void			*err_data;
 	size_t			err_data_size;
+	void			*auth_key;
+	size_t			auth_key_size;
```

Comment:
Libfabric 2.0 thought. To maintain ABI compat, each provider needs to check API version to select the correct version of fi_cq_err_entry. I am finding myself updating multiple providers with this check. While not difficult, seems like if fi_cq_readerr() was an external symbol, a compat symbol could be defined to do this translation in a single place. Thoughts?

Reply:
See ofi_cq_readerr() in util_cq.c.  Yes, if we change a data structure, we're forced to check the API version that the app has opened the fabric with to know the size of the structure that was passed in.

Since fi_cq_readerr() is not an external symbol, but a static inline, all pre-compiled apps would be using the cq ops function pointer and the original sized structure.  Adding a new external symbol doesn't help.

You may be able to avoid changing providers that use the util_cq, except for those that want to use the new functionality.  For example, you can add a new ofi_cq_write_error_trunc2() call that adds the src_addr.  ofi_cq_write_error_trunc() would call v2.  Existing providers are unchanged.  But ones that want to report the src_addr can use v2.  Obviously, the readerr() call still needs to handle things on the application side.